RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -186,9 +186,39 @@ - (NSDictionary *)customDirectEventTypes
                             
                             RCTMGLAnnotation *pin = [[RCTMGLAnnotation alloc] initWithLocationRightCallout:CLLocationCoordinate2DMake(coordinate.latitude, coordinate.longitude) title:title subtitle:subtitle id:id rightCalloutAccessory:imageButton];
                             [pins addObject:pin];
+                            
+                            if ([anObject objectForKey:@"annotationImage"]) {
+                                NSObject *annotationImage = [anObject valueForKey:@"annotationImage"];
+                                NSString *annotationImageURL = [annotationImage valueForKey:@"url"];
+                                CGFloat height = (CGFloat)[[annotationImage valueForKey:@"height"] floatValue];
+                                CGFloat width = (CGFloat)[[annotationImage valueForKey:@"width"] floatValue];
+                                if (!height || !width) {
```

Comment:
Can you make this optional? If an explicit width or height isn't given, grab the intrinsic dimensions from the UIImage.